packages/core/src/Asset/util/isValidHandle.ts

```diff
@@ -1,4 +1,26 @@
+/* eslint-disable unicorn/better-regex */
+// From ADA Handle Discord: slightly modified to not allow uppercase characters
+const REGEX_HANDLE = new RegExp(/^[a-z0-9_.-]{1,15}$/);
+// From ADA Handle Discord
+const REGEX_SUB_HANDLE = new RegExp(/(?:^[a-z0-9_.-]{1,15}$)|(?:^(?!.{29})[a-z0-9_.-]+@[a-z0-9_.-]{1,15}$)/g);
+
+/**
+ * From ADA Handle FAQ:
+ * Alphanumeric: [a-z][0-9]
+ * Dash: -
+ * Underscore: _
+ * Period: .
+ *
+ * Since handles are case-insensitive, this function only allows lowercase.
+ * Max 1-15 characters (or 3-31 characters for a subhandle)
+ */
 export const isValidHandle = (handle: string) => {
-  const pattern = /^[\w,.\-]*@{0,1}[\w,.\-]+$/;
-  return pattern.test(handle);
+  // 'g' modifier makes it stateful
+  REGEX_SUB_HANDLE.lastIndex = 0;
+  return (
+    REGEX_HANDLE.test(handle) ||
+    REGEX_SUB_HANDLE.test(handle) ||
+    // Pipe | in general is not valid, but an exception exists in mainnet
+    handle === '0|0'
+  );
 };
```

Comment:
I just wanted to mention that there were some cases in my career where I shot myself in the foot with schema validation, as in colleagues push data to database without appropriate checks. To be fair this is a simple schema with one entrypoint

Anyway, even tho it is considered a bad practice I like to enforce it in the database with constraint/check syntax or with triggers

In this case it would be trigger on insert which would set `is_valid` column to true/false depending on the handle itself


Just a suggestion, and in this case I don't think that its necessary due to simplicity of the schema. I just would like you guys to consider this in the future